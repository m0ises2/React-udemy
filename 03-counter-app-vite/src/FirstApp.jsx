import PropTypes from "prop-types";

export const FirstApp = ( { title, subtitle, age } ) => {
    return (
        <>
            <h1 data-testid='test title'> { title } </h1>
            <p data-testid='test subtitle'> { subtitle } </p>
            <p> { age } years old </p>
        </>
    ) // Esto es un fragmento
}

FirstApp.propTypes = {
    age: PropTypes.number.isRequired,
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    age: 35,
    subtitle: 'Subtitulo por defecto'
};
