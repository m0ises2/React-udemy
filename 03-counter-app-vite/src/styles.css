html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}