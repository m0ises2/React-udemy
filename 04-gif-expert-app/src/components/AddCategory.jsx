import { useState } from "react"
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState('');

    const onInputChange = ({ target: { value: inputValue } } ) => {
        setinputValue(inputValue)
    };

    const onFormSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());

        setinputValue('');
    };

    return (
        <form 
            aria-label="form"
            onSubmit={ onFormSubmit }>
            <input
                type="text"
                placeholder = "Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />            
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
