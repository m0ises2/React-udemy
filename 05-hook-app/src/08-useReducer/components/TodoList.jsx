import PropTypes from "prop-types";
import { TodoItem } from './TodoItem';

export const TodoList = ({ todos, onDeleteTodo, onToggleTodo }) => {
  return (
    <ul className='list-group'>
      {
        todos.map(todo => (
          <TodoItem
            key={ todo.id }
            id={ todo.id }
            description={ todo.description }
            done={ todo.done }
            onDeleteTodo={onDeleteTodo}
            onToggleTodo={onToggleTodo}
          />
        ))
      }
    </ul>
  )
};

TodoList.propTypes = {
    todos: PropTypes.array.isRequired
};
