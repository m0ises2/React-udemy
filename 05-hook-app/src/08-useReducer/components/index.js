export * from './AddTodo';
export * from './TodoItem';
export * from './TodoList';