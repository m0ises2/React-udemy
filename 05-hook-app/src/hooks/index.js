export * from './UseCounter';
export * from './useFetch';
export * from './useForm';