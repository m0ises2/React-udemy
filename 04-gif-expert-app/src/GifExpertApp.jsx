import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);
  
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    
    setCategories([newCategory.toLowerCase(),...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      
      {/* Lista de Gif */}
      { 
        categories.map(category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
        {/* Gif Item */}

    </>
  );
};
