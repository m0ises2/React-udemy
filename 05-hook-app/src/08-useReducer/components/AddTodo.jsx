import PropTypes from "prop-types";
import { useForm } from "../../hooks";

export const AddTodo = ({ onNewTodo }) => {
  const { description, onInputChange, onReset } = useForm({
    description: ''
  });

  const onSubmit = (event) => {
    event.preventDefault();

    if (description.trim().length <= 1) return;

    onNewTodo({
      id: new Date().getTime(),
      description,
      done: false
    });

    onReset();
  };

  return (
    <form
        onSubmit={ onSubmit }
    >
        <input 
          type="text" 
          placeholder='QuÃ© hay que hacer?' 
          className='form-control'
          name="description"
          value={ description }
          onChange={ onInputChange }
        />  
    <   button type='submit' className='btn btn-outline-primary mt-1'>  Agregar </button>
    </form>
  )
}

AddTodo.propTypes = {
    onNewTodo: PropTypes.func.isRequired
}