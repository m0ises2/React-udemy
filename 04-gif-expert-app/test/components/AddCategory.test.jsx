import { fireEvent, render, screen } from '@testing-library/react';
import { AddCategory } from "../../src//components";

describe('Pruebas de AddCategory', () => {
    const inputValue = 'A new Category this is';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<AddCategory onNewCategory={ () => {} } />);

        const input = screen.getByRole('textbox');
        
        fireEvent.input(input, {target: { value: inputValue } } );

        expect(container).toMatchSnapshot();
    });

    test('debe cambiar el valor de la caja de texto', () => {
        render(<AddCategory onNewCategory={ () => {} } />);
        
        const input = screen.getByRole('textbox');
        
        fireEvent.input(input, {target: { value: inputValue } } );

        expect(input.value).toBe(inputValue);
    });

    test('debe llamar onNewCategory si el input tiene un valor valido', () => {
        const onNewCAtegoryMock = jest.fn();
        
        render(<AddCategory onNewCategory={ onNewCAtegoryMock } />);
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: { value: inputValue } } );
        fireEvent.submit(form);
        
        expect(input.value).toBe('');
        expect(onNewCAtegoryMock).toHaveBeenCalledWith(inputValue);
        expect(onNewCAtegoryMock).toHaveBeenCalledTimes(1);
    });

    test('no debe llamar a onNewCategory si el valor es un string vacio', () => {
        const onNewCAtegoryMock = jest.fn();

        render(<AddCategory onNewCategory={ onNewCAtegoryMock } />);

        const form = screen.getByRole('form');
        
        fireEvent.submit(form);

        expect(onNewCAtegoryMock).not.toHaveBeenCalled();
    });
});
