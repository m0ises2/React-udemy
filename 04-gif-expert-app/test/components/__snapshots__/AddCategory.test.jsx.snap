// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AddCategory debe hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value="A new Category this is"
    />
  </form>
</div>
`;
